bitmark-references/bitmark-node-software/bitmark-blockchain.md

```diff
@@ -0,0 +1,173 @@
+  adds 7 more bits and the values is in Little-Endian form. The 9^th^
+  byte adds all eight bits bringing the total to 64.
+
+String
+: A sequence of UTF-8 byte sequence preceded by a VarInt byte count.
+  The count is validated by various routines to prevent massive
+  storage use.
+
+Binary
+: A sequence of bytes preceded by a VarInt byte count.  The count is
+  validated by various routines to prevent massive storage use.
+
+Map
+: Sequence of NUL separated UTF-8 byte sequences preceded by a VarInt
+  byte count.  There must be an even number sequences and these form
+  key→value pairs.  The must not be a trailing NUL byte.  In JSON
+  representation the NUL byte will show as `\u0000`.  The count is
+  validated by various routines to prevent massive storage use.
+
+Account
+: The encoded Ed25519 public key, there is a type code in this to
+  allow for other key types and it is preceeded by a VarInt byte
+  count.  Currently the count will be 33.
+
+Signature
+: An Ed25519 signature over all previous bytes including VarInt
+  fields.  This will preceeded by a VarInt that will be 64.  This
+  value will change once other signature algorithms are used and will
+  depend on the type code inside the Account field.
+
+
+## Transactions: Asset, Issue and Transfer Records
+
+### Asset Record
+
+This contains the metadata to describe an asset that will be issued on
+the blockchain.  All assets will have at least one issue.
+
+The Asset Record structure
+
+Item              Type       Description
+----------------  ---------  ------------------------
+Type code         Byte       Asset Record type code
+Name              String     Short descriptive name
+Fingerprint       String     A unique identifier for the Asset
+Metadata          Map        Key-Value to describe Asset
+Registrant        Account    The public key of the signer
+Signature         Signature  Ed25519 signature of signer
+
+
+The Issue Record structure
+
+Item              Type       Description
+----------------  ---------  ------------------------
+Type code         Byte       Issue Record type code
+AssetId           Binary     Link to asset record (SHA3-512 of asset fingerprint)
+Owner             Account    The public key of the signer
+Nonce             VarInt     Allow for multiple issues of the same asset
+Signature         Signature  Ed25519 signature of signer
+
+The Transfer Record structure
+
+Item              Type       Description
+----------------  ---------  ------------------------
+Type code         Byte       Transfer Record type code
+Link              Binary     Link to previous Issue or Transfer (SHA3-256)
+Owner             Account    The public key of the signer
+Signature         Signature  Ed25519 signature of linked previous owner
+CounterSignature  Signature  Ed25519 signature of this record owner
+
+
+## Transactions: Share
+
+
+## Transactions: Block ownership record
+
+
+## Blockchain structure
+
```

Comment:
It seems like this should be "Block Structure", not "Blockchain Structure"

We need an introduction here. Something like:

Transactions are gathered together into blocks, which contain a header of block information and Merkle tree containing hashes of all the transactions.